Analytics/Classes/Internal/SEGHTTPClient.m

```diff
@@ -51,7 +51,7 @@ - (NSURLSession *)sessionForWriteKey:(NSString *)writeKey
             @"Authorization" : [@"Basic " stringByAppendingString:[[self class] authorizationHeader:writeKey]],
             @"User-Agent" : [NSString stringWithFormat:@"analytics-ios/%@", [SEGAnalytics version]],
         };
-        session = [NSURLSession sessionWithConfiguration:config];
+        session = [NSURLSession sessionWithConfiguration:config delegate:self.httpSessionDelegate delegateQueue:NULL];
```

Comment:
1. Do we need to pass `delegateQueue:NULL`?
2. What's the default value for `session.delegate` before our change?

Reply:
Hi @ujjawalgarg 
1. We do need to pass `delegateQueue:NULL` but it can be set to NULL 
[see reference](https://developer.apple.com/documentation/foundation/nsurlsession/1411597-sessionwithconfiguration?language=objc)
2. by default `session.delegate` is null